command/update/command.go

```diff
@@ -269,8 +201,48 @@ func (c *Command) execute() error {
 			return microerror.MaskAny(err)
 		}
 
-		c.logger.Log("debug", fmt.Sprintf("updated endpoint for service '%s'", f.Kubernetes.Cluster.Service))
+		c.logger.Log("debug", fmt.Sprintf("added IP to endpoint of service '%s'", f.Kubernetes.Cluster.Service))
 	}
 
+	// Listen to OS signals issued by the Kubernetes scheduler.
+	listener := make(chan os.Signal, 2)
+	signal.Notify(listener, syscall.SIGTERM, syscall.SIGKILL)
+
+	c.logger.Log("debug", "waiting for termination signal")
+
+	s := <-listener
+
+	c.logger.Log("debug", fmt.Sprintf("received termination signal: %#v (%s)\n", s, s))
+
+	// Use the updater to actually delete the endpoints identified by the provided
+	// flags.
+	go func() {
+		action := func() error {
+			err := newUpdater.Delete(f.Kubernetes.Cluster.Namespace, f.Kubernetes.Cluster.Service, podInfos)
+			if err != nil {
+				return microerror.MaskAny(err)
+			}
+
+			return nil
+		}
+
+		err := backoff.Retry(action, backoff.NewExponentialBackOff())
+		if err != nil {
+			c.logger.Log("error", fmt.Sprintf("%#v", microerror.MaskAny(err)))
+			os.Exit(1)
+		}
+
+		c.logger.Log("debug", fmt.Sprintf("removed IP from endpoint of service '%s'", f.Kubernetes.Cluster.Service))
+		c.logger.Log("debug", "shutting down with exit code 0")
+
+		os.Exit(0)
```

Comment:
I'd rather like to pass cancelling context here and cancel after receiving from `listener`. That way you don't have to use `os.Exit` here which feels wrong in a background goroutine.

Reply:
I am not sure how you want to do this. Maybe provide some example code.

Reply:
I can tinker with that in the next pr.